src/ckeditor.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global CKEDITOR */
+
+import { getEditorNamespace } from './helpers.js';
+
+export default {
+	name: 'ckeditor',
+
+	render( createElement ) {
+		return createElement( 'div', {}, [
+			createElement( this.tagName )
+		] );
+	},
+
+	props: {
+		value: {
+			type: String,
+			default: ''
+		},
+		type: {
+			type: String,
+			default: 'classic',
+			validator: type => type in { classic: 1, inline: 1 }
+		},
+		editorUrl: {
+			type: String,
+			default: 'https://cdn.ckeditor.com/4.11.4/standard/ckeditor.js'
+		},
+		config: {
+			type: Object,
+			default: () => {}
+		},
+		tagName: {
+			type: String,
+			default: 'textarea'
+		},
+		readOnly: {
+			type: Boolean,
+			default: null // Use the null for default value, so `config.readOnly` can take precedence.
+		}
+	},
+
+	readOnly() {
+		return this.readOnly === null
+			? this.instance.readOnly
+			: this.readOnly;
+	},
```

Comment:
Using default value would break `config.readOnly`. Component attribute should take precedence, but still it should be possible to use config option. It is common that some project are rewritten with new frameworks, but they might use existing editor config.

Reply:
Ok, I see the issue now. I was thinking about updating `this.config` but not sure if it will make it more readable. Something like:
```Object.assign( {}, this.config, { readOnly: this.readOnly } )```
to avoid changing user object. In that case you could just use simple `readonly() { return this.instance.readOnly } )` also without setting this mode later on `instanceReady`.

Reply:
Slightly changing your idea, and we can get rid of `readOnly` getter. It looks like readability will be improved even further.